modules/detection/VADTalkMutedDetection.js

```diff
@@ -0,0 +1,292 @@
+     * needs to implement the following functions:
+     * - <tt>getSampleLength()</tt> - Returns the sample size accepted by getSampleLength.
+     * - <tt>getRequiredPCMFrequency()</tt> - Returns the PCM frequency at which the processor operates.
+     * - <tt>calculateAudioFrameVAD(pcmSample)</tt> - Process a 32 float pcm sample of getSampleLength size.
+     * @constructor
+     */
+    constructor(conference, createVADProcessor) {
+        super();
+
+        /**
+         * Member function that instantiates a VAD processor.
+         */
+        this._createVADProcessor = createVADProcessor;
+
+        /**
+         * Current {@link TrackVADEmitter}. VAD Emitter uses a {@link JitsiLocalTrack} and VAD processor to generate
+         * period voice probability scores.
+         */
+        this._vadEmitter = null;
+
+        /**
+         * Flag which denotes the current state of the detection service i.e.if there is already a processing operation
+         * ongoing.
+         */
+        this._processing = false;
+
+        /**
+         * Buffer that keeps the VAD scores for a period of time.
+         */
+        this._scoreArray = [];
+
+        /**
+         * Promise used to chain create and destroy operations associated with TRACK_ADDED and TRACK_REMOVED events
+         * coming from the conference.
+         * Because we have an async created component (VAD Processor) we need to make sure that it's initialized before
+         * we destroy it ( when changing the device for instance), or when we use it from an external point of entry
+         * i.e. (TRACK_MUTE_CHANGED event callback).
+         */
+        this._vadInitTracker = null;
+
+
+        this._processVADScore = this._processVADScore.bind(this);
+
+        /**
+         * {@link JitsiConference} bindings.
+         */
+        conference.on(JitsiConferenceEvents.TRACK_ADDED, this._trackAdded.bind(this));
+        conference.on(JitsiConferenceEvents.TRACK_REMOVED, this._trackRemoved.bind(this));
+        conference.on(JitsiConferenceEvents.TRACK_MUTE_CHANGED, this._trackMuteChanged.bind(this));
+
+        // TODO do we need to handle the case where tracks are removed, make sure this cleans up properly so
+        // we don't have any leeks i.e. stale JitsiLocalTracks
+    }
+
+    /**
+     * Determine if the current score is high enough that we should start the final score processing, and make sure
+     * there isn't already a process operation ongoing.
+     *
+     * @param {number} score - PCM sample VAD score.
+     * @return {boolean}
+     */
+    _shouldStartVADCompute(vadScore) {
+        return vadScore > VAD_VOICE_LEVEL && !this._processing;
+    }
+
+    /**
+     * Determine if the computed score over the configured timestamp should trigger an event.
+     *
+     * @param {number} computedScore - Computed VAD score.
+     * @returns {boolean} - Should or shouldn't trigger.
+     */
+    _shouldTriggerNotification(computedScore) {
```

Comment:
Well the reason I do this is so that it's easier to read and understand the overall flow, not for reuse.
So for instance, when you read _shouldStartVADCompute in a function you know what that logical condition inside does without even needing comments (the principle of self documenting code), as opposed to just reading a logical condition which you have to figure out.
But I guess that's subjective and is a matter coding style. I'll change.